synapse/storage/databases/main/cache.py

```diff
@@ -107,7 +111,9 @@ def get_all_updated_caches_txn(txn):
             "get_all_updated_caches", get_all_updated_caches_txn
         )
 
-    def process_replication_rows(self, stream_name, instance_name, token, rows):
+    def process_replication_rows(
+        self, stream_name: str, instance_name: str, token: int, rows: Iterable[Any]
+    ) -> None:
```

Comment:
Note that the `Iterable[Any]` seems to match what we do for the other instances of `process_replication_rows`, it isn't ideal but I think we want them to all match?